contracts/lib/Utilities.sol

```diff
@@ -69,18 +75,82 @@ library Utilities {
       length++;
       i = i >> 4;
     }
+
+    if (length > (size*2)) {
+      return "00";
+    }
+
+    bytes memory result = new bytes(size*2);
     // parse byte by byte and construct the string
     i = payload;
     uint mask = 15;
-    bytes memory result = new bytes(length);
-    uint k = length - 1;
+    uint k = (size*2) - 1;
 
     while (i != 0) {
       uint curr = (i & mask);
       result[k--] = curr > 9 ? byte(55 + uint8(curr)) : byte(48 + uint8(curr));
       i = i >> 4;
     }
 
+    uint j;
+    while (j < ((size*2)-length)) {
+      result[j++] = byte(uint8(48));
+    }
+
     return string(result);
   }
+
+  /**
+   * @dev Removes the last element of byte array
+   * @param payload bytes of data
+   * @return byte array without last element
+   */
+  function removeLastElement(
+    bytes memory payload
+  )
+  internal
+  pure
+  returns (
+    bytes memory
+  )
+  {
+    bytes memory output = new bytes(payload.length-1);
+    for (uint i = 0; i<payload.length-1; i++) {
```

Comment:
Couldn't find a better way to extract the first n bytes out of the payload. If you think there is a more efficient way(maybe bit shifting...) let me now.

Reply:
If n is fixed you could use some thing like in the extractIndex method. Can n be any number?

Reply:
in theory, this method can be reused making `n` any number, but in practice for us is 66 bytes.

Reply:
It makes sense. Why do we need a bool and bytes32 in the return? Isn't the bytes32 sufficient and it return 0 when it fails?

Reply:
I believe you are referring to the below method. I thought it was easier to do an assertion on a boolean than a bytes32 value with `00000...`. If you think that returning a `0` is cleaner, and easier to assert then I can change that.

Reply:
So on chain you can use require(bytesN != 0); The problem will if you use something like  web3 and it will return 0x000...(). ` but you can convert that to a number and compare it to 0